kotlin-ktor/src/main/java/com/baeldung/graphql/server/KtorApplication.kt

```diff
@@ -0,0 +1,44 @@
+package com.baeldung.graphql.server
+
+import com.baeldung.graphql.server.api.AttendeeMutation
+import com.baeldung.graphql.server.api.ConferenceMutation
+import com.baeldung.graphql.server.api.ConferenceQuery
+import com.baeldung.graphql.server.api.ConferenceSubscription
+import com.expediagroup.graphql.generator.SchemaGeneratorConfig
+import com.expediagroup.graphql.generator.TopLevelObject
+import com.expediagroup.graphql.generator.hooks.FlowSubscriptionSchemaGeneratorHooks
+import com.expediagroup.graphql.generator.toSchema
+import com.expediagroup.graphql.server.ktor.GraphQL
+import com.expediagroup.graphql.server.ktor.graphQLPostRoute
+import com.expediagroup.graphql.server.ktor.graphQLSubscriptionsRoute
+import graphql.schema.idl.SchemaPrinter
+import io.ktor.serialization.jackson.*
+import io.ktor.server.application.*
+import io.ktor.server.engine.*
+import io.ktor.server.netty.*
+import io.ktor.server.routing.*
+import io.ktor.server.websocket.*
+import java.time.Duration
+
+fun main() {
+    embeddedServer(Netty, port = 8080, module = Application::graphQLModule).start(wait = true)
+}
+
+fun Application.graphQLModule() {
+    install(WebSockets) {
+        pingPeriod = Duration.ofSeconds(1)
+        contentConverter = JacksonWebsocketContentConverter()
+    }
+    install(GraphQL) {
+        schema {
+            packages = listOf("com.baeldung.graphql.server")
+            queries = listOf(ConferenceQuery())
+            mutations = listOf(ConferenceMutation(), AttendeeMutation())
+            subscriptions = listOf(ConferenceSubscription())
+        }
```

Comment:
Out of curiosity, is there any support for classpath scanning of these instead of having to manually list them?

Reply:
Also, according to https://opensource.expediagroup.com/graphql-kotlin/docs/schema-generator/schema-generator-getting-started/

I was expecting to see a toSchema call using the graphql-kotlin-schema-generator

Reply:
Looks like the current PR actually does not use graphql-kotlin-schema-generator, if I comment out that dependency the build is still green

Reply:
Have a look at the task `graphqlGenerateSDL` https://opensource.expediagroup.com/graphql-kotlin/docs/plugins/gradle-plugin-tasks

I added in `build.gradle.kts`:
```
graphql {
    schema {
        packages = listOf("com.baeldung.graphql.server")
    }
}
```

This currently fails with:
```
Execution failed for task ':graphqlGenerateSDL'.
> There was a failure while executing work items
   > A failure occurred while executing com.expediagroup.graphql.plugin.gradle.actions.GenerateSDLAction
      > Schema requires all subscriptions to be public and return a valid type from the hooks. ConferenceSubscription has PUBLIC visibility modifier. The function return type is Flow
```

I was not able to figure out how to resolve it, but after commenting out the `ConferenceSubscription` entirely, it did correctly generate `build/schema.graphql` for the remaining types.

If we can figure out how to resolve this failure for the subscription, we can then stop hard coding the `src/main/resources/schema.graphql` file and replace it with the generated one.

It should also be possible to order the `./gradlew build` tasks in such a way that `graphqlGenerateSDL` is called before `graphqlGenerateClient`.

Reply:
✅Left a discussion on Jira